File: fun.dart
import 'dart:io';

void main() {
  stdout.write('Enter the first number: ');
  String? num1Str = stdin.readLineSync();
  double? num1 = double.tryParse(num1Str ?? '');

  stdout.write('Enter an operator (+, -, *, /): ');
  String? operator = stdin.readLineSync();

  stdout.write('Enter the second number: ');
  String? num2Str = stdin.readLineSync();
  double? num2 = double.tryParse(num2Str ?? '');

  if (num1 == null || num2 == null || operator == null || operator.length != 1) {
    print('Error: Invalid input.');
    return;
  }

  double? result;
  switch (operator) {
    case '+':
      result = num1 + num2;
      break;
    case '-':
      result = num1 - num2;
      break;
    case '*':
      result = num1 * num2;
      break;
    case '/':
      if (num2 == 0) {
        print('Error: Cannot divide by zero.');
        return;
      }
      result = num1 / num2;
      break;
    default:
      print('Error: Invalid operator.');
      return;
  }

  print('Result: $result');
}

File: fun.rs
use std::io::{self, Write};

fn main() {
    let mut num1_str = String::new();
    let mut op_str = String::new();
    let mut num2_str = String::new();

    print!("Enter the first number: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut num1_str).expect("Failed to read line");
    let num1: f64 = num1_str.trim().parse().expect("Please type a number!");

    print!("Enter an operator (+, -, *, /): ");
    io::stdout().flush().unwrap();
    io.stdin().read_line(&mut op_str).expect("Failed to read line");
    let operator = op_str.trim().chars().next().expect("Please enter an operator");

    print!("Enter the second number: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut num2_str).expect("Failed to read line");
    let num2: f64 = num2_str.trim().parse().expect("Please type a number!");

    let result = match operator {
        '+' => num1 + num2,
        '-' => num1 - num2,
        '*' => num1 * num2,
        '/' => {
            if num2 == 0.0 {
                println!("Error: Cannot divide by zero.");
                return;
            }
            num1 / num2
        }
        _ => {
            println!("Error: Invalid operator.");
            return;
        }
    };

    println!("Result: {}", result);
}
